cmd/utils/flags.go

```diff
@@ -737,24 +747,37 @@ func setIPC(ctx *cli.Context, cfg *node.Config) {
 	}
 }
 
-// SetULCFromFile setup ULC config from file if given.
-func SetULCFromFile(ctx *cli.Context, cfg *eth.Config, p2pCfg *p2p.Config) {
-	path := ctx.GlobalString(ULCModeConfigFlag.Name)
-	if path == "" {
+// SetULC setup ULC config from file if given.
+func SetULC(ctx *cli.Context, cfg *eth.Config) {
+	//ulc config haven't loaded from global config, ULC Config and ULC Trusted Nodes not defined
+	if cfg.ULC==nil && (ctx.GlobalIsSet(ULCModeConfigFlag.Name) || ctx.GlobalIsSet(ULCTrustedNodesFlag.Name)) == false  {
 		return
 	}
+	cfg.ULC=&eth.ULCConfig{}
 
-	cfgData, err := ioutil.ReadFile(path)
-	if err != nil {
-		Fatalf("Failed to read ULC config file: %v", err)
+	path := ctx.GlobalString(ULCModeConfigFlag.Name)
+	if path != "" {
+		cfgData, err := ioutil.ReadFile(path)
+		if err != nil {
+			Fatalf("Failed to read ULC config file: %v", err)
+		}
+
+		err = json.Unmarshal(cfgData, &cfg.ULC)
+		if err != nil {
+			Fatalf(err.Error())
+		}
 	}
 
-	err = json.Unmarshal(cfgData, &cfg.ULC)
-	if err != nil {
-		Fatalf(err.Error())
+	if trustedNodes:=ctx.GlobalString(ULCTrustedNodesFlag.Name); trustedNodes!="" {
+		cfg.ULC.ULCTrustedNodes=strings.Split(trustedNodes,";")
```

Comment:
Could it be that nodes passed by flag could be added to those passed by configuration file before instead replacing them? Also most flags split lists with a comma. Any reason to use a semicolon here?